### PR TITLE
clarify docker login

### PR DESCRIPTION
**Pull request by sallyom**
Signed-off-by: Sally O'Malley somalley@redhat.com

### CODE REVIEW

File: man/docker-login.1.md [hunk 1]
```diff
@@ -18,9 +18,11 @@ Register or log in to a Docker Registry located on the specified
 do not specify a `SERVER`, the command uses Docker's public registry located at
 `https://registry-1.docker.io/` by default.  To get a username/password for Docker's public registry, create an account on Docker Hub.
 
+`docker login` requires you to use `sudo` or be `root`.
```
Comment (ljrittle) [inline man/docker-login.1.md]: This is not a true statement, at least not in general across all `docker client` configurations.  I.e. when using docker to connect to a `docker-machine` provisioned `docker engine`.

Comment (sallyom) [inline man/docker-login.1.md]: Are there any other examples of docker client configurations that wouldn't require sudo with docker login?  I think the man page should at least specify that under most circumstances the credentials will be written in /root, no?

Comment (vbatts) [inline man/docker-login.1.md]: This assumption of being root is following in the theme of a limited user _not_ being in the host group of `docker`. 
Though, if the user decides against that security best-practice, and adds their user to the `docker` group, then sudo or root is not strictly required.

**Comment by rhatdan** [comment]
We should document both use cases.  Basically the content ends up in the homedir of the UID that is running the docker client.  If the docker client is being run via sudo or as root it will end up in /root directory.

**Comment by vbatts** [comment]
Correct

**Comment by SvenDowideit** [comment]
this info should be added to `docs/reference/commandline/login.md` as well.

File: man/docker-login.1.md [hunk 2]
```diff
@@ -18,9 +18,15 @@ Register or log in to a Docker Registry located on the specified
...
 
+`docker login` requires user to use `sudo` or be `root`, except when: 
+
+1.  using docker to connect to a `docker-machine` provisioned `docker engine`.
+2.  user is added to the `docker` group.  Note this will impact the security of your system; the `docker` group is `root` equivalent.  See [Docker Daemon Attack Surface](https://docs.docker.com/articles/security/#docker-daemon-attack-surface) for details. 
+
 You can log into any public or private repository for which you have
 credentials.  When you log in, the command stores encoded credentials in
 `$HOME/.docker/config.json` on Linux or `%USERPROFILE%/.docker/config.json` on Windows.
+**Note**:  When running `sudo docker login` or `docker login` as `docker` group member, credentials are saved in `/root/.docker/config.json`.
```
Comment (thaJeztah) [inline man/docker-login.1.md]: I don't think it's stored in `/root/...` if ran as a user thats in the `docker` group, only when ran using `sudo`.

Also, could you add a blank line before the note and format it in our standard "note" formatting;

```Markdown
> **Note**: ......
>
``

File: man/docker-login.1.md [hunk 3]
```diff
@@ -18,9 +18,15 @@ Register or log in to a Docker Registry located on the specified
 do not specify a `SERVER`, the command uses Docker's public registry located at
 `https://registry-1.docker.io/` by default.  To get a username/password for Docker's public registry, create an account on Docker Hub.
 
+`docker login` requires user to use `sudo` or be `root`, except when: 
+
+1.  using docker to connect to a `docker-machine` provisioned `docker engine`.
```
Comment (thaJeztah) [inline man/docker-login.1.md]: I think technically, this applies to any remote daemon

**Comment by sallyom** [comment]
@thaJeztah @SvenDowideit addressed your feedback, anything else?  Thanks.

**Comment by thaJeztah** [comment]
LGTM! thanks @sallyom

**Comment by SvenDowideit** [comment]
LGTM \o/ :)

**Pull request merged**